data_iterator.go

```diff
@@ -225,7 +109,7 @@ func (d *DataIterator) Run() {
 					}
 
 					logger.Debugf("updated last successful PK to %d", pkpos)
-					d.CurrentState.UpdateLastSuccessfulPK(table.String(), pkpos)
+					d.StateTracker.UpdateLastSuccessfulPK(table.String(), pkpos)
```

Comment:
copy pasting the other comment 😛 

is there a test that would fail if we removed this line?

how about one that would fail if it was executed at the wrong time?

Reply:
This is more easily testable, but I think black box testing via the ruby integration tests should be enough. 

As stated earlier, I also use the ruby integration test to assert the dumped data iterator coordinates, but requires additional investigation and can't be done in this PR as the ruby test framework is not merged yet.